### PR TITLE
Blocking of actions cause you have no power here

### PR DESCRIPTION
**Pull request by Nightbaka**

### CODE REVIEW

File: frontend/src/components/AuthProvider/AuthProvider.tsx
Comment (Ninja00Shadow) [inline frontend/src/components/AuthProvider/AuthProvider.tsx:12]: I’ll leave the decision up to you, but I think it would be better to ignore the user part in user_status and just save the permissions. It would make everything clearer.

File: frontend/src/api/auth/authStatus.ts
Comment (Ninja00Shadow) [inline frontend/src/api/auth/authStatus.ts:15]: I think it would be better to return just the permissions. I talked with Samuel, and the user in userStatus isn’t needed on the frontend. It will remain in the API because it might be useful for other apps and isn’t a significant issue. However, we it just causes unnecessary confusion in our app.

File: frontend/src/functions/authorization.ts
Comment (Ninja00Shadow) [inline frontend/src/functions/authorization.ts:90]: Let me know what you think—some of these might be better than yours, though not all.

```suggestion
type Operation = "c" | "r" | "u" | "d" | "l";

function getEffectivePermissions(permissions: UserStatus['permissions'], namespace: string, resource: string): Operation[] {
    const nsPermissions = permissions[namespace] || permissions["*"] || {};
    const resPermissions = nsPermissions[resource] || nsPermissions["*"] || [];
    return resPermissions as Operation[];
}

export function hasPermission(userStatus: UserStatus, namespace: string, resource: string, operation: Operation): boolean {
    const resPermissions = getEffectivePermissions(userStatus.permissions, namespace, resource);
    return resPermissions.includes(operation);
}

export function hasPermissionInAnyNamespace(userStatus: UserStatus, resource: string, operation: Operation): boolean {
    const permissions = userStatus.permissions;
    return Object.keys(permissions).some((namespace) => {
        const resPermissions = getEffectivePermissions(permissions, namespace, resource);
        return resPermissions.includes(operation);
    });
}

export function hasAnyPermissionInAnyNamespace(userStatus: UserStatus, resource: string): boolean {
    const permissions = userStatus.permissions;
    return Object.keys(permissions).some((namespace) => {
        const resPermissions = getEffectivePermissions(permissions, namespace, resource);
        return resPermissions.length > 0;
    });
}

export function hasPermissionInAnyResource(userStatus: UserStatus, namespace: string, operation: Operation): boolean {
    const nsPermissions = userStatus.permissions[namespace] || userStatus.permissions["*"];
    if (!nsPermissions) return false;

    return Object.keys(nsPermissions).some((resource) => {
        const resPermissions = getEffectivePermissions(userStatus.permissions, namespace, resource);
        return resPermissions.includes(operation);
    });
}

export function allowedNamespaces(userStatus: UserStatus, operation: Operation, resource: string): string[] {
    const permissions = userStatus.permissions;
    const namespaces = new Set<string>();

    Object.keys(permissions).forEach((namespace) => {
        const resPermissions = getEffectivePermissions(permissions, namespace, resource);
        if (resPermissions.includes(operation)) {
            namespaces.add(namespace);
        }
    });

    return Array.from(namespaces);
}

export function allowedResources(userStatus: UserStatus, operation: Operation, namespace: string): string[] {
    const permissions = userStatus.permissions;
    const resources = new Set<string>();

    const nsPermissions = permissions[namespace] || {};
    const wildcardNsPermissions = permissions["*"] || {};

    const combinedPermissions = { ...wildcardNsPermissions, ...nsPermissions };

    Object.keys(combinedPermissions).forEach((resource) => {
        const resPermissions = getEffectivePermissions(permissions, namespace, resource);
        if (resPermissions.includes(operation)) {
            resources.add(resource);
        }
    });

    return Array.from(resources);
}

export function allowedOperations(userStatus: UserStatus, namespace: string, resource: string): Operation[] {
    return getEffectivePermissions(userStatus.permissions, namespace, resource);
}
```

File: frontend/src/api/auth/authStatus.ts
Comment (Nightbaka) [reply frontend/src/api/auth/authStatus.ts:15]: Will do

File: frontend/src/functions/authorization.ts
Comment (Nightbaka) [reply frontend/src/functions/authorization.ts:90]: I agree some of these are cleaner code, however I see some bugs in some of the refactored ones you shared. To accept this refactor I would have to break my mind on these and analize if there are any edge cases I might have overlooked. Can we agree to disagree - I'll add the operation type, and some of the smaller refactors, but most of the functions will stay the same. If you see any fault in my logic or way I am using javascript please point out the specific case

File: frontend/src/pages/ResourcePage.tsx
Comment (Nightbaka) [reply frontend/src/pages/ResourcePage.tsx:50]: I disagree if somehow the userState is somehow undefined on frontend we don't want to block users actions - this is not a safety concern as the backend will verify the action anyways. This should never happen but I prefer to keep like this anyways. Will remove onclick check its not necessary

**Review by vgoriukhina — COMMENTED** [review]
Considering David's comments, the key points have been outlined. Removing userStatus would require significant changes.  In the future, resource details will need additional constraints, so it’s better to address the userStatus issue upfront.

File: frontend/src/components/AuthProvider/AuthProvider.tsx
Comment (Nightbaka) [reply frontend/src/components/AuthProvider/AuthProvider.tsx]: yes

**Comment by Nightbaka** [comment]
All addressed and pushed. I haven't tested the last commit yet cause my env broke, but will do before merge. Please re-review

File: frontend/src/hooks/useListReleases.ts
Comment (Ninja00Shadow) [inline frontend/src/hooks/useListReleases.ts:13]:
```suggestion
        if (permissions && !hasPermissionInAnyNamespace(permissions, "Helm", "l")) {
```

**Review by Ninja00Shadow — COMMENTED** [review]
I'll give you approve after tests.

File: frontend/src/pages/ResourcePage.tsx
Comment (Ninja00Shadow) [reply frontend/src/pages/ResourcePage.tsx]: I have no idea what you are talking about.

**Comment by Nightbaka** [comment]
> I nealy overlooked this. You didn't authorize helm actions. You athorized only list and not rollback and uninstall.

Done

**Review by Ninja00Shadow — APPROVED** [review]
Please also change appVersion.

**Pull request merged**
